diff -Naur gcc-7.3.0.old/gcc/ada/gcc-interface/Makefile.in gcc-7.3.0/gcc/ada/gcc-interface/Makefile.in
--- gcc-7.3.0.old/gcc/ada/gcc-interface/Makefile.in	2018-12-13 17:49:43.001927878 +0200
+++ gcc-7.3.0/gcc/ada/gcc-interface/Makefile.in	2018-12-13 17:53:01.615721223 +0200
@@ -1932,7 +1932,7 @@
 endif
 
 # ARM linux, GNU eabi
-ifeq ($(strip $(filter-out arm% linux-gnueabi%,$(target_cpu) $(target_os))),)
+ifeq ($(strip $(filter-out arm% linux-gnueabi% linux-muslgnueabi%,$(target_cpu) $(target_os))),)
   LIBGNAT_TARGET_PAIRS = \
   a-intnam.ads<a-intnam-linux.ads \
   s-inmaop.adb<s-inmaop-posix.adb \
diff -Naur gcc-7.3.0.old/gcc/ada/s-taprop-linux.adb gcc-7.3.0/gcc/ada/s-taprop-linux.adb
--- gcc-7.3.0.old/gcc/ada/s-taprop-linux.adb	2018-12-13 17:49:42.989927767 +0200
+++ gcc-7.3.0/gcc/ada/s-taprop-linux.adb	2018-12-13 17:53:01.615721223 +0200
@@ -271,14 +271,19 @@
 
          begin
             --  Set the rwlock to prefer writer to avoid writers starvation
+            --  Comment the prefere stuff out
+            --  As this is not portable to musl
+            --  (_np stands for non-portable and is a gnu clib extr)
 
             Result := pthread_rwlockattr_init (RWlock_Attr'Access);
             pragma Assert (Result = 0);
 
-            Result := pthread_rwlockattr_setkind_np
-              (RWlock_Attr'Access,
-               PTHREAD_RWLOCK_PREFER_WRITER_NONRECURSIVE_NP);
-            pragma Assert (Result = 0);
+            --  snip snip
+            --  Result := pthread_rwlockattr_setkind_np
+            --    (RWlock_Attr'Access,
+            --    PTHREAD_RWLOCK_PREFER_WRITER_NONRECURSIVE_NP);
+            --  pragma Assert (Result = 0);
+            --  snip snip
 
             Result := pthread_rwlock_init (L.RW'Access, RWlock_Attr'Access);
 
diff -Naur gcc-7.3.0.old/gcc/ada/terminals.c gcc-7.3.0/gcc/ada/terminals.c
--- gcc-7.3.0.old/gcc/ada/terminals.c	2018-12-13 17:49:42.997927841 +0200
+++ gcc-7.3.0/gcc/ada/terminals.c	2018-12-13 17:53:01.615721223 +0200
@@ -1061,12 +1061,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 
-/* On some system termio is either absent or including it will disable termios
-   (HP-UX) */
-#if !defined (__hpux__) && !defined (BSD) && !defined (__APPLE__) \
-  && !defined (__rtems__)
-#   include <termio.h>
-#endif
 
 #include <sys/ioctl.h>
 #include <termios.h>
@@ -1112,7 +1106,7 @@
 #if defined (__APPLE__) || defined (BSD)
 #define USE_OPENPTY
 #elif defined (__linux__)
-#define USE_GETPT
+#define USE_OPENPTY
 #elif defined (__sun__)
 #define USE_CLONE_DEVICE "/dev/ptmx"
 #elif defined (_AIX)
diff -Naur gcc-7.3.0.old/gnattools/Makefile.in gcc-7.3.0/gnattools/Makefile.in
--- gcc-7.3.0.old/gnattools/Makefile.in	2018-12-13 17:49:42.377922068 +0200
+++ gcc-7.3.0/gnattools/Makefile.in	2018-12-13 17:53:01.615721223 +0200
@@ -52,7 +52,7 @@
 ADA_CFLAGS=@ADA_CFLAGS@
 
 # Variables for gnattools.
-ADAFLAGS= -gnatpg -gnata
+ADAFLAGS= -gnatpg -gnata -gnatwn
 
 # For finding the GCC build dir, which is used far too much
 GCC_DIR=../gcc
